src/Nemo.jl

```diff
@@ -423,6 +426,16 @@ const FiniteField = FlintFiniteField
 const RealField = ArbField
 const ComplexField = AcbField
 
+###############################################################################
+#
+#   Internal/External definitions
+#
+###############################################################################
+
+div(a, b) = Base.div(a, b)
+
+divrem(a, b) = Base.divrem(a, b)
```

Comment:
If I understand correctly, should this not be in AbstractAlgebra?

Reply:
It might still work if we move it there. I am really not sure.

The purpose of it is so that Nemo's version of divrem, will use the Base definition (which we have defined in Nemo) when there isn't something more specific defined (e.g. fmpz).

For example, we define Base.div for power series in Nemo (because we can't export Nemo's div to the user any other way). In order for Nemo to use that same definition internally, these redirect it to there.

Reply:
I see. But this redirect should also be done for all non-`BigInt` types in AbstractAlgebra I think, and this should cover also all that is happening here in Nemo. We are already doing it here:
https://github.com/Nemocas/AbstractAlgebra.jl/blob/6fa064a1e5cc8eabbb549fcbeae20055e0fac13a/src/AbstractAlgebra.jl#L64-L70

Reply:
In answer to your question: no. These are already in AbstractAlgebra, and they don't work for Nemo when they are only there.

On the other hand it's possible that another change I made makes them redundant. I'll try removing them and see if the tests still pass.

Reply:
If they don't work, then something is off I would say. If you remove them in Nemo, what does `@which AbstractAlgebra.divrem(fmpq(1), fmpq(1))` say?

Reply:
In fact the tests pass if I remove these two. I think I wasn't overloading the AbstractAlgebra definitions previously, which is why I inserted these. Now that I'm doing that, these are actually redundant, it seems. I'll remove them and push momentarily.

Reply:
Fixed.